lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/security/FileBasedAccessControl.java

```diff
@@ -606,9 +605,13 @@ public void checkCanExecuteTableProcedure(ConnectorSecurityContext context, Sche
     @Override
     public void checkCanExecuteFunction(ConnectorSecurityContext context, FunctionKind functionKind, SchemaRoutineName function)
     {
-        if (functionKind == TABLE) {
-            denyExecuteFunction(function.toString());
+        switch (functionKind) {
+            case SCALAR, AGGREGATE, WINDOW:
+                return;
+            case TABLE:
+                denyExecuteFunction(function.toString());
```

Comment:
Add 
>  // TODO implement

?

Reply:
`Make checkCanExecuteFunction checks future proof` is meant to be just a code refactor.

Reply:
You won. 😄 
It was not obvious from squashed commits PR view.